### DISCUSSION TITLE
Welcome to ib-gateway-docker Discussions!

### DISCUSSION THREAD

**Discussion by gnzsnz** [gnzsnz/ib-gateway-docker#14] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Anggel7x** [comment]
I'm not sure how to run the TWS instead of the IB Gateway, I cannot find it in the documentation or the specific bash file.

**Comment by gnzsnz** [comment]
Hi

It's a different image, for example for tws it's
https://github.com/gnzsnz/ib-gateway-docker/pkgs/container/tws-rdesktop

I created different docker-compose files for each case. For tws you can use
https://github.com/gnzsnz/ib-gateway-docker/blob/master/tws-docker-compose.yml

Hope it helps.

Regards

**Reply by Anggel7x** [reply]
Thank you so much, with that I was able to deploy TWS. However, whenever I try to connect my python app with the TWS I get this error and I'm not able to get any information from my account therefore the app doesn't work.

18:01:56 socat[2325] E connect(5, AF=2 127.0.0.1:7497, 16): Connection refused

**Comment by gnzsnz** [comment]
That line is saying that socat is working. this is good, but TWS api is not reachable, not good

> 18:01:56 socat[2325] E connect(5, AF=2 127.0.0.1:7497, 16): Connection refused

make sure API is enabled in TWS, settings > api > settings > Enable active X and socket clients
that should fix it. if you also set "use/store settings on server" you should be able to get this from IBKR everythime.

you can set TWS_SETTINGS_PATH (set it inside the /config volume, ex /config/tws_settings) this will save settings on disk.

There are way to enable the TWS API from the container scripts, but I can avoid all those extra steps and failure points, by just doing a click on the GUI. So it's not there. pull requests are always welcome.

**Reply by Anggel7x** [reply]
I was able to activate it through IBC command server, however since it kept refusing the conection from telnet I used the commandsend.sh inside the container. I wrote a shell file that does this:


#!/bin/bash

docker exec -it <container_id> bash -c 'cd /opt/ibc && ./commandsend.sh ENABLEAPI'

After that I had no problem.

Thank you so much!
